pkgs/development/libraries/arrow-cpp/thrift.patch

```diff
@@ -0,0 +1,16 @@
+diff --git a/cmake_modules/FindThrift.cmake b/cmake_modules/FindThrift.cmake
+index 07028971d..87f2fce35 100644
+--- a/cmake_modules/FindThrift.cmake
++++ b/cmake_modules/FindThrift.cmake
+@@ -135,11 +135,6 @@ find_package_handle_standard_args(
+   HANDLE_COMPONENTS)
+ 
+ if(Thrift_FOUND)
+-  if(ARROW_THRIFT_USE_SHARED)
+-    add_library(thrift::thrift SHARED IMPORTED)
+-  else()
+-    add_library(thrift::thrift STATIC IMPORTED)
+-  endif()
```

Comment:
I'm looking into upstreaming this patch.

Reply:
Why is this a duplicate target? Is it somehow finding Thrift's cmake MODULE before getting here?

Reply:
It is. I'm not yet entirely sure how.

Reply:
Maybe, try looking at output with `cmakeFlags = [ "--trace"`

Reply:
Thanks. Add `--trace` was helpful.

It looks like arrow invokes `find_package(Thrift)`

```
arrow-cpp> /build/apache-arrow-9.0.0/cpp/cmake_modules/ThirdpartyToolchain.cmake(257):  find_package(${FIND_PACKAGE_ARGUMENTS} )
```

which invokes `thrift`'s `ThriftConfig.cmake`:

```
arrow-cpp> /nix/store/0g5hv59vjfh7yfd2csymp9czznis7y73-thrift-0.16.0/lib/cmake/thrift/ThriftConfig.cmake(53):  if(NOT TARGET thrift::thrift )
```

and that invokes `thriftTargets.cmake` which creates the `thrift::thrift` target:

```
arrow-cpp> /nix/store/0g5hv59vjfh7yfd2csymp9czznis7y73-thrift-0.16.0/lib/cmake/thrift/thriftTargets.cmake(54):  add_library(thrift::thrift STATIC IMPORTED )
```

Some time after that creation of a duplicate target `thrift::thrift` is attempted by arrow:

```
arrow-cpp> /build/apache-arrow-9.0.0/cpp/cmake_modules/FindThrift.cmake(139):  add_library(thrift::thrift SHARED IMPORTED )
```

Reply:
Then one may need to ensure that calls to `find_package` have consistent `MODULE`/`CONFIG` switches.
https://cmake.org/cmake/help/latest/command/find_package.html

Reply:
I'm not sure what exactly that means. Would you say this is an issue in Arrow or Thrift or something else?

Reply:
I mean that if arrow-cpp does (first three trace snippets)
```
find_package(Thrift CONFIG)
```
following by (last trace snippet)
```
find_package(Thrift MODULE)
```
then it's a bug in arrow-cpp.

And, if the comment by the `"-DThrift_SOURCE=AUTO"` is accurate, we were not supposed to use cmake modules for Thrift, but find it with pkg-config. I guess they switched up the things again.